tests/nfs_backend_sc_retain_test.go

```diff
@@ -0,0 +1,218 @@
+				ObjectMeta: metav1.ObjectMeta{
+					Name: scName,
+					Annotations: map[string]string{
+						string(mayav1alpha1.CASTypeKey):   "nfsrwx",
+						string(mayav1alpha1.CASConfigKey): string(casObjStr),
+					},
+				},
+				Provisioner: "openebs.io/nfsrwx",
+			})
+			Expect(err).To(BeNil(), "while creating SC{%s}", scName)
+		})
+	})
+
+	When(fmt.Sprintf("pvc with storageclass %s is created", scName), func() {
+		It("should create a pvc ", func() {
+			By("building a pvc")
+			pvcObj, err := pvc.NewBuilder().
+				WithName(pvcName).
+				WithNamespace(applicationNamespace).
+				WithStorageClass(scName).
+				WithAccessModes(accessModes).
+				WithCapacity(capacity).Build()
+			Expect(err).To(BeNil(), "while building pvc %s/%s object", applicationNamespace, pvcName)
+
+			By("creating above pvc")
+			err = Client.createPVC(pvcObj)
+			Expect(err).To(BeNil(), "while creating pvc %s/%s", applicationNamespace, pvcName)
+
+			_, err = Client.waitForPVCBound(pvcName, applicationNamespace)
+			Expect(err).To(BeNil(), "while waiting %s/%s pvc to bound", applicationNamespace, pvcName)
+
+			pvcObj, err = Client.getPVC(applicationNamespace, pvcName)
+			Expect(err).To(BeNil(), "while fetching pvc %s/%s", applicationNamespace, pvcName)
+
+			backendPvcName = "nfs-" + pvcObj.Spec.VolumeName
+			_, err = Client.waitForPVCBound(backendPvcName, openebsNamespace)
+			Expect(err).To(BeNil(), "while waiting %s/%s pvc to bound", openebsNamespace, backendPvcName)
+
+			pvcObj, err = Client.getPVC(openebsNamespace, backendPvcName)
+			Expect(err).To(BeNil(), "while fetching pvc %s/%s", applicationNamespace, pvcName)
+
+			backendPvName = pvcObj.Spec.VolumeName
+		})
+	})
+
+	When(fmt.Sprintf("pvc with storageclass %s is deleted", scName), func() {
+		It("should delete the pvc", func() {
+			By(fmt.Sprintf("pvc with storageclass %s is deleted", scName))
+			err := Client.deletePVC(applicationNamespace, pvcName)
+			Expect(err).To(BeNil(), "while deleting pvc %s/%s", applicationNamespace, pvcName)
+
+			maxRetryCount := 10
+			isPvcDeleted := false
+			for retries := 0; retries < maxRetryCount; retries++ {
+				_, err := Client.getPVC(applicationNamespace, pvcName)
+				if err != nil && k8serrors.IsNotFound(err) {
+					isPvcDeleted = true
+					break
+				}
+				time.Sleep(time.Second * 5)
+			}
+			Expect(isPvcDeleted).To(BeTrue(), "pvc should be deleted")
+
+			isBackendPvcDeleted := false
+			for retries := 0; retries < maxRetryCount; retries++ {
+				_, err := Client.getPVC(openebsNamespace, backendPvcName)
+				if err != nil && k8serrors.IsNotFound(err) {
+					isBackendPvcDeleted = true
+					break
+				}
+				time.Sleep(time.Second * 5)
+			}
+			Expect(isBackendPvcDeleted).To(BeTrue(), "backend pvc should be deleted")
+		})
+	})
+
+	When("verify backend PV state after application PVC deletion", func() {
+		It("should have backend PV in released state", func() {
+			Expect(backendPvName).NotTo(BeNil(), "backend PV name should not be empty")
```

Comment:
Good to compare with BeEmpty() instead of BeNil()
```suggestion
			Expect(backendPvName).NotTo(BeEmpty(), "backend PV name should not be empty")
```